doc/usage/extensions/intersphinx.rst

```diff
@@ -162,21 +167,50 @@ linking:
 
    The default value is an empty list.
 
-   When a cross-reference without an explicit inventory specification is being
-   resolved by intersphinx, skip resolution if it matches one of the
-   specifications in this list.
+   When a non-:rst:role:`external` cross-reference is being resolved by
+   intersphinx, skip resolution if it matches one of the specifications in this
+   list.
 
    For example, with ``intersphinx_disabled_reftypes = ['std:doc']``
    a cross-reference ``:doc:`installation``` will not be attempted to be
-   resolved by intersphinx, but ``:doc:`otherbook:installation``` will be
-   attempted to be resolved in the inventory named ``otherbook`` in
+   resolved by intersphinx, but ``:external:otherbook+doc:`installation``` will
+   be attempted to be resolved in the inventory named ``otherbook`` in
    :confval:`intersphinx_mapping`.
    At the same time, all cross-references generated in, e.g., Python,
    declarations will still be attempted to be resolved by intersphinx.
 
-   If ``*`` is in the list of domains, then no references without an explicit
-   inventory will be resolved by intersphinx.
+   If ``*`` is in the list of domains, then no non-:rst:role:`external`
+   references will be resolved by intersphinx.
+
+Explicitly Reference External Objects
+-------------------------------------
+
+The Intersphinx extension provides the following role.
+
+.. rst:role:: external
+
+   .. versionadded:: 4.4
+
+   Use Intersphinx to perform lookup only in external projects, and not the
+   current project. Intersphinx still needs to know the type of object you
+   would like to find, so the general form of this role is to write the
+   cross-refererence as if the object is in the current project, but then prefix
+   it with ``:external``.
+   The two forms are then
+
+   - ``:external:domain:reftype:`target```,
+     e.g., ``:external:py:class:`zipfile.ZipFile```, or
+   - ``:external:reftype:`target```,
+     e.g., ``:external:doc:`installation```.
+
+   If you would like to constrain the lookup to a specific external project,
+   then the key of the project, as specified in :confval:`intersphinx_mapping`,
+   is added as well to get the two forms 
 
+   - ``:external:invname+domain:reftype:`target```,
+     e.g., ``:external:python+py:class:`zipfile.ZipFile```, or
```

Comment:
When I saw this notation for the first time, I felt using `+` sign is cool. But this example makes my opinion opposite.

In my eyes, `:external:python+doc:` is good, but `:external:python+py:class:` is not so good. I guess the `+` sign confuses me. So I found 3 parts `external`, `python+py` and `class` from the latter example.

How about changing this notation?

* `:external+python:py:class:`
  *  Add a invname as a suffix of the "external" part.
  * It's almost similar to the current idea.
* `:external@python:py:class:`
  * Add a invname as a suffix of the "external" part using a `@` sign.
  * `@` sign is usually used to represent a location because it's equivalent to "at".
  * It looks like a bit complicated.

Reply:
I agree, the ``+`` looks like it has "higher precedence" than ``:`. I like ``@`` but if I read Docutils correctly (https://github.com/sphinx-doc/sphinx/pull/9062#issuecomment-879861369) then we can't use it. Only ``-._+`` can be used.
I think your first suggestion with ``:external+python:py:class:`` is the best. Just for the visual picture, the full set of variations are then
- ``:external+python:py:class:``, specific inventory, specific domain,
- ``:external:py:class:``, all inventories, specific domain,
- ``:external+python:class:``, specific inventory, default domain,
- ``:external:class:``, all inventories, default domain.

Reply:
The PR is now updated with this new syntax.